WIP: Added httplug basic implementation

Techworker opened the pull request:
Added httplug, this is still work in progress.

Renamed Request and Response classes in the RemoteApi NS to Action and Result, as Response and Request as names are related to actual HTTP requests and responses. This will hopefully prevent getting too confused.

I adjusted the AttachToTangleTest for now.

Techworker committed:
Added httplug basic implementation. Will need more work but it's a start.

Renamed or Request and Response classes in the RemoteApi NS to Action
and Result, as Response and Request as names are related to actual HTTP
requests and responses. This will hopefully help getting too confused.

robinlehrmann commented on composer.json:
```diff
@@ -29,16 +29,24 @@
+        "phpdocumentor/reflection-docblock": "^4.2",
+        "php-http/mock-client": "^1.0",
+        "php-http/message": "^1.0",
+        "guzzlehttp/psr7": "^1.0"
```
why do you need guzzle in test environment ?

Techworker replied:
I followed the implementation guide at http://docs.php-http.org/en/latest/httplug/library-developers.html - I guess the mock does not include a functionality to create responses / requests.

robinlehrmann replied:
You don't need to create an instance of it. Take a look at this: http://docs.php-http.org/en/latest/clients/mock-client.html
You should mock the request or response. Maybe with prophezie?

Techworker replied:
Adjusted.

robinlehrmann commented on composer.json:
```diff
@@ -29,16 +29,24 @@
         "psr/container": "^1.0",
-        "kornrunner/keccak": "^1.0"
+        "kornrunner/keccak": "^1.0",
+        "psr/http-message": "^1.0",
```
you don't need it due php-http/httplug has already a dependency on psr/http-message

Techworker replied:
I followed the implementation guide at http://docs.php-http.org/en/latest/httplug/library-developers.html - if we don't need id, it can go.

robinlehrmann replied:
The documentation can be outdated. If you take a look in the composer.json of http-plug you will see, that the dependency is already defined in it.

Techworker replied:
Ok, adjusted.

Techworker committed:
Removed psr/http-message as it's already a dependency of php-http/httplug.
Removed the guzzle message implementation guzzlehttp/psr7 from the require-dev.

Techworker committed:
Adjusted test to new implementation.

Techworker commented:
Merge without much testing, but that's accepted for now.
